### PR TITLE
test: fix flaky redis unit tests

### PR DESCRIPTION
**Pull request by Uzlopak**
We were running the unit tests in a very flaky way. Instead of actually waiting the redis flushall command we were having a noop callback. So it could happen, that flushall was not done and we were already running the next test.

This PR fixes that by implementing a teardown for those redis tests and fixing the corresponding assertions. 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/route-rate-limit.test.js [hunk 1]
```diff
@@ -335,18 +337,20 @@ test('Skip on redis error', async t => {
   t.equal(res.headers['x-ratelimit-limit'], 2)
   t.equal(res.headers['x-ratelimit-remaining'], 1)
 
-  redis.flushall(noop)
-  await redis.quit()
```
Comment (Eomm) [inline test/route-rate-limit.test.js:339]: the scope of this test is to verify the `skipOnError` flag, so without quitting the redis connection we are not testing it

File: test/global-rate-limit.test.js [hunk 2]
```diff
@@ -358,13 +357,16 @@ test('With redis store', async t => {
...
   t.equal(res.headers['x-ratelimit-remaining'], 1)
   t.equal(res.headers['x-ratelimit-reset'], 1)
+
+  t.teardown(async () => {
+    await redis.flushall()
+    await redis.quit()
+  })
 })
 
 test('Skip on redis error', async t => {
```
Comment (climba03003) [inline test/global-rate-limit.test.js:372]: This test is doing what it suppose to do. We need another way to simulate Redis error.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: test/global-rate-limit.test.js (hunk 2 above)
Comment (Uzlopak) [reply test/global-rate-limit.test.js:372]: I patched it now according to @Eomm remarks. 

Is it not acceptable?

Comment (climba03003) [reply test/global-rate-limit.test.js:372]: I believe we are just commenting on the same thing.
The patch appear later than my comment.

**Review by Eomm — APPROVED** [review]

**Pull request merged**